### PR TITLE
IXDS target selection dialog for multi-target GUI loading

### PR DESCRIPTION
**Pull request by hermfischer**
When loading an IXDS with multiple targets by GUI without ixdsTarget parameter a dialog is provided to select the desired target.

#### Reason for change
Prior operation required a formula parameter or command line operation.

#### Description of change
Dialog is provided to select target on loading multi-target IXDS from GUI without ixdsTarget parameter.

Documentation in header comment of plugin inlineDocumentSet.py needs to be transferred to arelle.org/pub documentation page(s).

Multi-target operation requires plugin inlineDocumentSet (auto-loaded by EdgarRenderer, validate/EFM and validate/ESEF plugins).

#### Steps to Test
IXDS test suite must still pass (of course!) (requires special test script setup as before this pull request).

Use multi-target single and multi-doc IXDSes, with resources in first or last doc, assure dialog works and formula parameter (tools->formula parameters) still work.

Use attached multi-target multi-doc test [multi-io.zip](https://github.com/Arelle/Arelle/files/10892704/multi-io.zip) to verify that with GUI interactive-user specification of target, which occurs only after all ix docs are loaded, that discovery only happens after this target selection.  (For scripted target selection the IXDS test suite is sufficient, because when target is known in advance of ix doc loading, discovery can proceed during ix doc loading vs after). 

 For ` --file '[{"ixds":[{"file":file1},{"file":file2}...],"ixdsTarget":"xyz"}]'` additional testing is suggested.

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Comment by hermfischer** [comment]
Adapted ixds test suite multi-io to require DTS discovery after GUI selection of target, only for interactive specification of target.  [multi-io.zip](https://github.com/Arelle/Arelle/files/10892704/multi-io.zip)

File: arelle/ModelDocument.py
Comment (austinmatherne-wk) [inline arelle/ModelDocument.py:1323]: The negated negative check is confusing:
```suggestion
        if any(pluginMethod(self.modelXbrl)
```

**Review by austinmatherne-wk — COMMENTED** [review]
linter is failing
> tox: lint
  lint: commands[0]> flake8 arelle
  arelle/plugin/inlineXbrlDocumentSet.py:24:1: W293 blank line contains whitespace
  arelle/plugin/inlineXbrlDocumentSet.py:648:1: W293 blank line contains whitespace
  arelle/plugin/inlineXbrlDocumentSet.py:664:87: W291 trailing whitespace
  lint: exit 1 (19.56 seconds) /home/runner/work/Arelle/Arelle> flake8 arelle pid=3284

File: arelle/ModelDocument.py
Comment (sagesmith-wf) [reply arelle/ModelDocument.py:1323]: Seems like maybe you want them all to be true? I would use `all` in that case

File: arelle/plugin/inlineXbrlDocumentSet.py
Comment (sagesmith-wf) [inline arelle/plugin/inlineXbrlDocumentSet.py:96]: Should this be checked for existence? I am also unclear why specifying `entrypoint` is mutually exclusive with specifying `ixdsTarget` if the `entrypoint.ixdsTarget` isn't set. I would expect something like this:

```python
_target = None

if 'ixdsTarget' in kwargs.get('entrypoint', {}):
    ...
elif 'ixdsTarget' in kwargs:
   ...
elif len(modelXbrl.modelManager.formulaOptions.parameterValues.get('ixdsTarget', [])`
   ...
modelXbrl.ixdsTarget = None if _target == '(default)' else _target or None
```

to avoid NPE's and provide fallbacks

File: arelle/ModelDocument.py
Comment (hermfischer) [reply arelle/ModelDocument.py:1323]: Changed to all

File: arelle/plugin/inlineXbrlDocumentSet.py
Comment (hermfischer) [reply arelle/plugin/inlineXbrlDocumentSet.py:96]: Originally if entrypoint is used, ixdsTarget was only specified for non-default targets and the default was the absence of ixdsTarget.  However if we want so respecify that ixdsTarget has to be "(default)" to recognize its use in entrypoint then the above structure makes more sense and the ixdsTarget in parameters can be used when no ixdsTarget in entrypoint.  Since this is probably not yet used by anybody except validation conformance suite (where it comes in kwargs), maybe we could or should change.  Discuss on stand up?  Code not changed today.

**Review by austinmatherne-wk — DISMISSED** [review]

**Review by sagesmith-wf — DISMISSED** [review]
Though lets discuss https://github.com/Arelle/Arelle/pull/604/files#r1127157276 briefly before merging imo

**Review by austinmatherne-wk — APPROVED** [review]

**Review by sagesmith-wf — APPROVED** [review]

**Review by austinmatherne-wk — APPROVED** [review]

**Review by derekgengenbacher-wf — APPROVED** [review]

**Comment by derekgengenbacher-wf** [comment]
QA +10 Testing locally succeeded. Was able to get the target selection screen successfully.

**Pull request merged**
